x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { generateKeyPairSync, createSign, randomBytes } from 'crypto';
+
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import type { EncryptedSavedObjectsClient } from '@kbn/encrypted-saved-objects-plugin/server';
+import { SECURITY_EXTENSION_ID } from '@kbn/core-saved-objects-server';
+
+import { MESSAGE_SIGNING_KEYS_SAVED_OBJECT_TYPE } from '../../constants';
+import { appContextService } from '../app_context';
+
+interface MessageSigningKeys {
+  private_key: string;
+  public_key: string;
+  passphrase: string;
+}
+
+export interface MessageSigningServiceInterface {
+  generateKeyPair(providedPassphrase?: string): Promise<void>;
+  sign(serializedMessage: Buffer | object): Promise<{ data: Buffer; signature: string }>;
+  getPublicKey(): Promise<string>;
+}
+
+export class MessageSigningService implements MessageSigningServiceInterface {
+  #esoClient: EncryptedSavedObjectsClient;
```

Comment:
I know we've got this in a few places now, but it does seem a little strange to me to use JavaScript's private property syntax when we have the more descriptive `private` keyword available to us in TypeScript. Maybe an interesting discussion to be had around using language primitives vs TypeScript features in the future.